lib/task/migrate/migrations/arMigration0184.class.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Update info related to menu locking
+ *
+ * @package    AccesstoMemory
+ * @subpackage migration
+ */
+class arMigration0184
+{
+  const
+    VERSION = 184, // The new database version
+    MIN_MILESTONE = 2; // The minimum milestone required
+
+  /**
+   * Upgrade
+   *
+   * @return bool True if the upgrade succeeded, False otherwise
+   */
+  public function up($configuration)
+  {
+    $lockedMenus = array(
+      'byName' => array(
+        'accessions',
+        'browseInstitution',
+        'clipboard',
+        'globalReplace',
+        'groups',
+        'importSkos',
+        'jobs',
+        'login',
+        'logout',
+        'myProfile',
+        'plugins',
+        'privacy',
+        'settings',
+        'staticPagesMenu',
+        'taxonomies',
+        'users'
+      ),
+      'byId' => array(
+        QubitMenu::ROOT_ID,
+        QubitMenu::BROWSE_ID,
+        QubitMenu::IMPORT_ID,
+        QubitMenu::MAIN_MENU_ID,
+        QubitMenu::MANAGE_ID,
+        QubitMenu::QUICK_LINKS_ID,
+        QubitMenu::ADD_EDIT_ID,
+        QubitMenu::ADMIN_ID
+      )
+    );
+
+    $setting = QubitSetting::getByName('menu_locking_info');
+    if (isset($setting))
```

Comment:
An edge case, since you're talking about adding this migration to a non fully upgraded site, if that site is on 2.4.x the setting won't exist. Not a problem is they are in 2.5.x already but you could create the setting if it doesn't exist to make the migration a bit more idempotent.